components/contact/ContactInputs.tsx

```diff
@@ -0,0 +1,114 @@
+"user client";
+
+import toast from "react-hot-toast";
+import emailjs from "@emailjs/browser";
+import { useRef, MutableRefObject, useState } from "react";
+
+import InputField from "@/components/contact/InputField";
+
+const ContactInputs = () => {
+  const form: MutableRefObject<HTMLFormElement | null> = useRef(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  // NOTE - Cannot make it exportable because of the use of IsLoading
+  const inputField = [
+    {
+      label: "What's your name?",
+      id: "name",
+      name: "name",
+      type: "text",
+      maxLength: 50,
+      required: true,
+    },
+    {
+      label: "What's your email?",
+      id: "email",
+      name: "email",
+      type: "email",
+      maxLength: 500,
+      required: true,
+    },
+    {
+      label: "Write something about your project goals and time frame",
+      id: "goalsTimeFrame",
+      name: "goalsTimeFrame",
+      type: "text",
+      maxLength: 500,
+      required: true,
+      isTextArea: true,
+    },
+    {
+      label: (
+        <>
+          How to reach out to you back?
+          <br className="block lg:hidden" />
+          <span className="text-black400">{` eg. phone number or email`}</span>
+        </>
+      ),
+      id: "howToReactOut",
+      name: "howToReactOut",
+      type: "text",
+      maxLength: 500,
+      pattern: "^(?:\\d{10}|\\w+@\\w+\\.\\w{2,3})$",
+      required: true,
+      // NOTE - Cannot make it exportable because of the use of IsLoading
+      isLoading,
```

Comment:
Is there a way you could get rid of isLoading and place it somewhere else so that you can make the inputField constant exportable? Put in other words, why do you need isLoading here?

Reply:
I can see a number of ways in which you could go about this.

1) Pull the reach out label out of the inputField const and treat it separately.
2) Create the isLoading property but set it up to a default value ex: true. Make the inputField const exportable and then import it into this component. Finally, create a state variable called reachOutLabel. Do the following :

```javascript
const [reachOutLabel. setReachOutLabel] = useState(inputField[3])
// then when you want to change the loading value
setReachOutLabel(prevReachOutLabel => {
    prevReachOutLabel.isLoading = false
    return prevReachOutLabel
})

```